peerdid/peer_did.py

```diff
@@ -0,0 +1,66 @@
+import re
+from typing import List, Optional
+
+from peerdid.storage import Storage, FileStorage
+from peerdid.types import PEER_DID, PublicKeyAgreement, PublicKeyAuthentication, JSON
+
+
+def is_peer_did(peer_did: PEER_DID) -> bool:
+    """
+    Checks if peer_did parameter matches Peer DID spec
+    (https://identity.foundation/peer-did-method-spec/index.html#matching-regex)
+    :param peer_did: peer_did to check
+    :return: true if peer_did matches spec, otherwise false
+    """
+    if peer_did is None:
+        return False
+    peer_did_pattern = re.compile(r'^did:peer:[01](z)([1-9a-km-zA-HJ-NP-Z]{46,47})$')
+    return bool(re.match(peer_did_pattern, peer_did))
+
+
+def create_peer_did_numalgo_0(inception_key: PublicKeyAgreement) -> PEER_DID:
+    """
+    Generates peer_did according to the zero algorithm
+    (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
+    For this type of algorithm did_doc can be obtained from peer_did
+    :param inception_key: the key that creates the DID and authenticates when exchanging it with the first peer
+    :return: generated peer_did
+    """
+    pass
+
+
+def create_peer_did_numalgo_2(encryption_keys: List[PublicKeyAgreement], signing_keys: List[PublicKeyAuthentication],
+                              service_endpoint: str) -> PEER_DID:
+    """
+    Generates peer_did according to the second algorithm
+    (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
+    For this type of algorithm did_doc can be obtained from peer_did
+    :param encryption_keys: list of encryption keys
+    :param signing_keys: list of signing keys
+    :param service_endpoint: string conforming to the DID specification (https://www.w3.org/TR/did-core/#services)
+    :return: generated peer_did
+    """
+    pass
+
+
+def resolve_peer_did(peer_did: PEER_DID) -> JSON:
```

Comment:
I believe we should keep an option version_id parameter there, so that we don't change the API signature once we support dynamic layers.
We just need to mention in docstrings that the version_id is not used for now, as we support only static layer where did doc never changes.